javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistTypeDeclarationAdapter.java

```diff
@@ -37,68 +40,81 @@
-    if (ctClass != null) {
-      Arrays.stream(ctClass.getDeclaredFields())
-          .forEach(f -> fieldDecls.add(new JavassistFieldDeclaration(f, typeSolver)));
-      try {
-        collectDeclaredFields(ctClass.getSuperclass(), fieldDecls);
-      } catch (NotFoundException e) {
-        // We'll stop here
-      }
+    public List<ResolvedTypeParameterDeclaration> getTypeParameters() {
+        if (null == ctClass.getGenericSignature()) {
+            return Collections.emptyList();
+        } else {
+            try {
+                SignatureAttribute.ClassSignature classSignature =
+                        SignatureAttribute.toClassSignature(ctClass.getGenericSignature());
+                return Arrays.stream(classSignature.getParameters())
+                        .map((tp) -> new JavassistTypeParameter(tp, JavassistFactory.toTypeDeclaration(ctClass, typeSolver), typeSolver))
+                        .collect(Collectors.toList());
+            } catch (BadBytecode badBytecode) {
+                throw new RuntimeException(badBytecode);
+            }
+        }
     }
-  }
 
-  public List<ResolvedTypeParameterDeclaration> getTypeParameters() {
-    if (null == ctClass.getGenericSignature()) {
-      return Collections.emptyList();
-    } else {
-      try {
-        SignatureAttribute.ClassSignature classSignature =
-            SignatureAttribute.toClassSignature(ctClass.getGenericSignature());
-        return Arrays.<SignatureAttribute.TypeParameter>stream(classSignature.getParameters())
-            .map((tp) -> new JavassistTypeParameter(tp, JavassistFactory.toTypeDeclaration(ctClass, typeSolver), typeSolver))
-            .collect(Collectors.toList());
-      } catch (BadBytecode badBytecode) {
-        throw new RuntimeException(badBytecode);
-      }
+    public Optional<ResolvedReferenceTypeDeclaration> containerType() {
+        try {
+            return ctClass.getDeclaringClass() == null ?
+                    Optional.empty() :
+                    Optional.of(JavassistFactory.toTypeDeclaration(ctClass.getDeclaringClass(), typeSolver));
+        } catch (NotFoundException e) {
+            throw new RuntimeException(e);
+        }
     }
-  }
 
-  public Optional<ResolvedReferenceTypeDeclaration> containerType() {
-    try {
-      return ctClass.getDeclaringClass() == null ?
-          Optional.empty() :
-          Optional.of(JavassistFactory.toTypeDeclaration(ctClass.getDeclaringClass(), typeSolver));
-    } catch (NotFoundException e) {
-      throw new RuntimeException(e);
+    /**
+     * Helper method to get the list of ancestors for the annotation.
+     *
+     * @return The list of ancestors.
+     */
+    public List<ResolvedReferenceType> getAncestors() {
+        return Collections.singletonList(
+                JavaParserFacade.get(typeSolver)
+                        .classToResolvedType(Annotation.class)
+                        .asReferenceType()
+        );
```

Comment:
Does `JavassistTypeDeclarationAdapter` only apply to annotations?

Just trying to get my head around how/why `getAncestors()` (plural) can return a singleton list.

I can only assume that getting the ancestor for e.g. `List` doesn't trigger this method?

Reply:
Looking into the code, looks like other implementations that use it (`JavassistEnumDeclaration`, `JavassistClassDeclaration` and `JavassistInterfaceDeclaration`) implement the same logic. It could probably be simplified by making them call this method.